game-core/src/main/java/games/strategy/engine/data/CompositeChange.java

```diff
@@ -22,6 +24,21 @@ public CompositeChange(final List<Change> changes) {
     this.changes = new ArrayList<>(changes);
   }
 
+  public CompositeChange flatten() {
+    return new CompositeChange(
+        changes.stream()
+            .map(
+                change -> {
+                  if (change instanceof CompositeChange) {
```

Comment:
nit: A ternary would be worth consider having here, would make the lambda syntax a bit nicer.